src/cli.rs

```diff
@@ -243,5 +256,16 @@ pub fn extract_args(matches: &ArgMatches) -> Config {
     args.txs_batch_size = matches
         .value_of("batch-size")
         .map(|batch_size_str| batch_size_str.parse().expect("can't parse batch-size"));
+
+    let (_, kp_auth_path) = ConfigInput::compute_keypair_path_setting(
+        matches.value_of("keeper-authority").unwrap_or(""),
+        &config.keypair_path,
+    );
+
+    if let Ok(kpa) = read_keypair_file(kp_auth_path) {
+        args.authority = kpa;
+    } else if matches.is_present("identity") {
+        panic!("could not parse identity path");
+    }
```

Comment:
Was the error message not updated after a copy/paste?
It seems like it should be talking about the `keeper-authority` value here.
Also, it is better to include context information in the error message.
The exact command line is not always at hand, when the error is analyzed.

```suggestion
    args.authority = match read_keypair_file(kp_auth_path)
        Ok(authority) => authority,
        Err(err) => panic!(
            "Could not read the \"keeper-authority\" keypair file:\n\
             file: {kp_auth_path}\n\
             error: {err}"
        ),
    };
```

It is still not ideal to use `panic!` for user visible errors.
As it prints the error in a way that is more suitable for internal programming errors.
But it seems this is the style here.

Reply:
good catch